R/flat-model-matrix-helpers.R

```diff
@@ -1,12 +1,28 @@
 
 
 #' Inefficiently transpose a sparse matrix
+#'
+#' Also transparently forwards to `t` for a dense matrix
 #' 
 #' @param m sparse matrix
 #' @return transpose of m
-st = function(m) Matrix::Matrix(t(as.matrix(m)))
+st = function(m) {
+  if (class(m) != 'matrix')
+    m = as.matrix(m)
```

Comment:
Is this to handle a case of a vector being passed instead of a matrix? Maybe consider adding a comment or printing a warning, since the conversion to a matrix type (untransposed)may lead to unexpected .

Reply:
It handles the case of a "matrix" or one of the "Matrix" types.  The "Matrix" types don't have a common sub-class so I don't test for it directly (there are a LOT of Matrix types).  Maybe there is an is.Matrix or something like that, I'll check.

Reply:
Nope, that doesn't exist.  Here we should only be getting one of the Matrix/matrix types so I don't think passing in a vector is an issue (just based on our usage) but it is a weakness in how the package is designed (formulas evaluated to build matrices and then merging the results).  It needs a better design to avoid these issues.